File: app/src/main/java/com/yawar/memo/fragment/CallHistoryFragment.java
package com.yawar.memo.fragment;

import android.os.Bundle;

import androidx.appcompat.widget.SearchView;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;

import com.tsuryo.swipeablerv.SwipeableRecyclerView;
import com.yawar.memo.R;
import com.yawar.memo.adapter.CallAdapter;
import com.yawar.memo.adapter.ChatRoomAdapter;
import com.yawar.memo.model.CallModel;
import com.yawar.memo.model.ChatRoomModel;
import com.yawar.memo.model.SearchRespone;
import com.yawar.memo.modelView.CallHistoryModelView;
import com.yawar.memo.modelView.ChatRoomViewModel;
import com.yawar.memo.sessionManager.ClassSharedPreferences;
import com.yawar.memo.utils.BaseApp;

import java.util.ArrayList;


public class CallHistoryFragment extends Fragment {

CallHistoryModelView callHistoryModelView;
    RecyclerView recyclerView;
    CallAdapter itemAdapter;
    SearchView searchView;
    ClassSharedPreferences classSharedPreferences;
    LinearLayout linerNoCalls;
    ProgressBar progressBar;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_call_history, container, false);
        recyclerView =  view.findViewById(R.id.recycler_view);
        linerNoCalls = view.findViewById(R.id.liner_no_call_history);
        progressBar = view.findViewById(R.id.progress_circular);

        recyclerView.setHasFixedSize(true);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(linearLayoutManager);
        classSharedPreferences = BaseApp.getInstance().getClassSharedPreferences();
        itemAdapter = new CallAdapter( getActivity());

        callHistoryModelView = new ViewModelProvider(this).get(CallHistoryModelView.class);
//        callHistoryModelView.loadData(classSharedPreferences.getUser().getUserId());
        callHistoryModelView.loadData(classSharedPreferences.getUser().getUserId()).observe(getActivity(), new Observer<ArrayList<CallModel>>() {
            @Override
            public void onChanged(ArrayList<CallModel> callModels) {
                ArrayList<CallModel> list = new ArrayList<>();
                if (callModels != null) {
                    System.out.println("no call");
                     if(callModels.isEmpty()){

                         linerNoCalls.setVisibility(View.VISIBLE);
                         recyclerView.setVisibility(View.GONE);
                     }
                     else {
                         linerNoCalls.setVisibility(View.GONE);
                         recyclerView.setVisibility(View.VISIBLE);

                         for (CallModel callModel : callModels) {
                             list.add(callModel.clone());
                         }
                         itemAdapter.setData((ArrayList<CallModel>) list);

                     }
                }
            }
            });


        recyclerView.setAdapter(itemAdapter);

        callHistoryModelView.loading.observe(getActivity(), new androidx.lifecycle.Observer<Boolean>() {
            @Override
            public void onChanged(Boolean aBoolean) {
                if (aBoolean!=null) {
                    System.out.println("loadinggggg");

                    if (aBoolean) {
                        recyclerView.setVisibility(View.GONE);
                        progressBar.setVisibility(View.VISIBLE);
                        linerNoCalls.setVisibility(View.GONE);

                    } else {
                        recyclerView.setVisibility(View.VISIBLE);
                        progressBar.setVisibility(View.GONE);
                        linerNoCalls.setVisibility(View.VISIBLE);


                    }
                }
            }
        });

        searchView = view.findViewById(R.id.search_view);
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;

            }

            @Override
            public boolean onQueryTextChange(String newText) {
                itemAdapter.getFilter().filter(newText);
                return false;
            }
        });


        return view;
    }


}

File: app/src/main/java/com/yawar/memo/fragment/CallHistoryFragment.kt
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.ProgressBar
import androidx.appcompat.widget.SearchView
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.yawar.memo.R
import com.yawar.memo.adapter.CallAdapter
import com.yawar.memo.call.RequestCallActivity
import com.yawar.memo.databinding.FragmentCallHistoryBinding
import com.yawar.memo.databinding.FragmentSearchBinding
import com.yawar.memo.model.CallModel
import com.yawar.memo.modelView.CallHistoryModelView
import com.yawar.memo.sessionManager.ClassSharedPreferences
import com.yawar.memo.utils.BaseApp


class CallHistoryFragment : Fragment(), CallAdapter.CallbackInterface {

    lateinit var callHistoryModelView: CallHistoryModelView
    lateinit var itemAdapter: CallAdapter
    lateinit var classSharedPreferences: ClassSharedPreferences
    lateinit var binding: FragmentCallHistoryBinding



    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_call_history,container,false)
        val view = binding.root
        binding.recyclerView.setHasFixedSize(true)
        val linearLayoutManager = LinearLayoutManager(context)
        linearLayoutManager.orientation = LinearLayoutManager.VERTICAL
        binding.recyclerView.layoutManager = linearLayoutManager
        classSharedPreferences = BaseApp.getInstance().classSharedPreferences
        itemAdapter = CallAdapter(this)
        callHistoryModelView = ViewModelProvider(this)[CallHistoryModelView::class.java]
        callHistoryModelView.loadData(classSharedPreferences.user.userId!!)
            .observe(
                requireActivity(),
                Observer<ArrayList<CallModel?>?> { callModels ->
                    val list = ArrayList<CallModel?>()
                    if (callModels != null) {
                        println("no call")
                        if (callModels.isEmpty()) {
                            binding.linerNoCallHistory.visibility = View.VISIBLE
                            binding.recyclerView.visibility = View.GONE
                        } else {
                            binding.linerNoCallHistory.visibility = View.GONE
                            binding.recyclerView.visibility = View.VISIBLE
                            for (callModel in callModels) {
                                if (callModel != null) {
                                    list.add(callModel.clone())
                                }
                            }
                            itemAdapter.setData(list)
                        }
                    }
                })
        binding.recyclerView.adapter = itemAdapter
        callHistoryModelView.loadingMutableLiveData.observe(
            requireActivity()
        ) { aBoolean ->
            if (aBoolean != null) {
                if (aBoolean) {
                    binding.recyclerView.visibility = View.GONE
                    binding.progressCircular.visibility = View.VISIBLE
                    binding.linerNoCallHistory.visibility = View.GONE
                } else {
                    binding.recyclerView.visibility = View.VISIBLE
                    binding.progressCircular.visibility = View.GONE
                    binding.linerNoCallHistory.visibility = View.VISIBLE
                }
            }
        }
        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                itemAdapter.filter.filter(newText)
                return false
            }
        })
        return view
    }

    override fun onHandleSelection(position: Int, callModel: CallModel?) {
        if (callModel != null) {
            val id : String
            if(callModel.answer_id == classSharedPreferences.user.userId){
                  id = callModel.caller_id.toString()
            }
            else{
                id = callModel.answer_id.toString()

            }

            if(callModel.call_type == "video"){
                val intent = Intent(requireActivity(), RequestCallActivity::class.java)
                //                Intent intent = new Intent(ConversationActivity.this, CompleteActivity.class);

                intent.putExtra("anthor_user_id", id)
                intent.putExtra("user_name", callModel.username)
                intent.putExtra("isVideo", true)
                intent.putExtra("fcm_token", "")
                intent.putExtra("image_profile", callModel.image)
                startActivity(intent)
            }
            else{
                val intent = Intent(requireActivity(), RequestCallActivity::class.java)
                //                Intent intent = new Intent(ConversationActivity.this, CompleteActivity.class);

                intent.putExtra("anthor_user_id", id)
                intent.putExtra("user_name", callModel.username)
                intent.putExtra("isVideo", false)
                intent.putExtra("fcm_token", "")
                intent.putExtra("image_profile", callModel.image)
                startActivity(intent)
            }
            }

        }
    }





